### DISCUSSION TITLE
Customize the public folder with a service provider or configuration

### DISCUSSION THREAD

**Discussion by Claudye** [laravel/framework#42900] [Ideas]
Studying the code of the framework I found that it is possible to customize the root folder of the laravel source code and I experimented and it worked, it's great.
The structure of the project that has become is:
rootDir
public
Editing the public folder works fine.

I have problems with the files that are in the public folder.
The php artisan serve commands; php artisan Storage:link, had problem.
When I searched through the code, the line` $this->instance('path.public', $this->publicPath());` shows that we can use a service provider or `setPublic `method, or use the configuration file to set the public path.
I tested with this line `$this->app->instance('path.public',__DIR__ .'/../../../public'); `in the service app service provider and it worked fine.

Laravel is really awesome.

This can allow the public folder to not always be called public, or even it can be a deep folder But it could also be a real security measure brought to laravel

**Comment by Claudye** [comment]
What I propose is that this be a default configuration of laravel

**Comment by tobya** [comment]
I dont understand what benefit is provided from being able to modify where the public folder is?  Perhaps I'm missing something but i dont see how this is a security issue.
